deepspeed/runtime/engine.py

```diff
@@ -1454,6 +1462,22 @@ def save_checkpoint(self, save_dir, tag=None, client_state={}, save_latest=True)
         if tag is None:
             tag = f"global_step{self.global_steps}"
 
+        if self.checkpoint_tag_validation_enabled():
```

Comment:
Maybe it'd be good to refactor the validation into a utility that could be reused and/or tested in isolation?